src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false) {
+    super(name);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, hungry) {
+    super(name);
+    this.hungry = hungry;
+  }
+
+  bite(animal) {
+    if (animal.hasOwnProperty('hidden')) {
+      if (!animal.hidden) {
+        animal.health -= 50;
```

Comment:
consider check the instance of animal and combine these conditions using and operator (&&)